app/validation.py

```diff
@@ -41,8 +42,14 @@ def load_schemas(schemas_path, schema_names):
 _SCHEMAS = load_schemas(JSON_SCHEMAS_PATH, SCHEMA_NAMES)
 
 
-def get_validator(schema_name):
-    schema = _SCHEMAS[schema_name]
+def get_validator(schema_name, enforce_required=True, required_fields=None):
+    if required_fields is None:
+        required_fields = []
+    if enforce_required:
+        schema = _SCHEMAS[schema_name]
+    else:
+        schema = _SCHEMAS[schema_name].copy()
```

Comment:
This only does a shallow copy, given that the schemas are nested dictionaries I think this should use [`copy.deepcopy`](https://docs.python.org/3.4/library/copy.html#copy.deepcopy)